src/PhpPact/Consumer/Registry/Interaction/Body/AbstractBodyRegistry.php

```diff
@@ -23,10 +23,10 @@ public function __construct(
 
     public function withBody(Text|Binary|Multipart $body): void
     {
-        $success = match ($body::class) {
-            Binary::class => $this->client->call('pactffi_with_binary_file', $this->interactionRegistry->getId(), $this->getPart(), $body->getContentType(), $body->getContents()->getValue(), $body->getContents()->getSize()),
-            Text::class => $this->client->call('pactffi_with_body', $this->interactionRegistry->getId(), $this->getPart(), $body->getContentType(), $body->getContents()),
-            Multipart::class => array_reduce(
+        $success = match (true) {
```

Comment:
I'm no PHP expert, but what's the point of having a `match (true)`? I understand this to be similar to an `if (true) {...}` which seems redundant. (Of course, I may be quite wrong here).

Reply:
```
$value = match ($x) {
    $y => $z,
};
```

means when `$x` strictly equals to `$y`, value of `$z` is return.

```
$value = match (true) {
    $y => $z,
};
```

is the same, when `$y` is a boolean, and its value is `true`, then `$z` is return.

So it's not the same as `if (true)`, but it's the same as `if ($y === true)`